### PR TITLE
feat: ConfigProvider support locale

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

ref #16911

### 💡 Background and solution

Merge LocaleProvider into ConfigProvider

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Merge LocaleProvider into ConfigProvider      |
| 🇨🇳 Chinese |    将 LocaleProvider 合并入 ConfigProvider       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/config-provider/demo/all.md](https://github.com/ant-design/ant-design/blob/locale/components/config-provider/demo/all.md)
[View rendered components/config-provider/index.en-US.md](https://github.com/ant-design/ant-design/blob/locale/components/config-provider/index.en-US.md)
[View rendered components/config-provider/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/components/config-provider/index.zh-CN.md)
[View rendered components/date-picker/index.en-US.md](https://github.com/ant-design/ant-design/blob/locale/components/date-picker/index.en-US.md)
[View rendered components/date-picker/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/components/date-picker/index.zh-CN.md)
[View rendered components/locale-provider/index.en-US.md](https://github.com/ant-design/ant-design/blob/locale/components/locale-provider/index.en-US.md)
[View rendered components/locale-provider/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/components/locale-provider/index.zh-CN.md)
[View rendered docs/react/faq.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/docs/react/faq.zh-CN.md)
[View rendered docs/react/getting-started.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/docs/react/getting-started.zh-CN.md)
[View rendered docs/react/i18n.en-US.md](https://github.com/ant-design/ant-design/blob/locale/docs/react/i18n.en-US.md)
[View rendered docs/react/i18n.zh-CN.md](https://github.com/ant-design/ant-design/blob/locale/docs/react/i18n.zh-CN.md)

### CODE REVIEW

File: components/locale-provider/ar_EG.tsx [hunk 1]
```diff
@@ -1,42 +1,3 @@
-import Pagination from 'rc-pagination/lib/locale/ar_EG';
-import DatePicker from '../date-picker/locale/ar_EG';
-import TimePicker from '../time-picker/locale/ar_EG';
-import Calendar from '../calendar/locale/ar_EG';
+import locale from '../locale/ar_EG';
```
Comment (afc163) [inline components/locale-provider/ar_EG.tsx]: 目录应该是 `locales`

Comment (zombieJ) [reply components/locale-provider/ar_EG.tsx]: 代码里用的目录是 `locale`：
* `'!components/*/locale/index.tsx'`
* `date-picker/locale`
* `moment/locale/zh-cn`

还是保持一致吧

Comment (afc163) [reply components/locale-provider/ar_EG.tsx]: 👌

File: docs/react/faq.zh-CN.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ import { Menu, Breadcrumb, Icon } from 'antd';
 
 ### 我的组件默认语言是英文的？如何切回中文的。
 
-请尝试使用 [LocaleProvider](https://ant.design/components/locale-provider-cn/) 组件来包裹你的应用。
+请尝试使用 [ConfigProvider](https://ant.design/components/config-provider-cn/) 组件来包裹你的应用。
```
Comment (afc163) [inline docs/react/faq.zh-CN.md]: 链接最好给到具体段落。

File: components/config-provider/index.zh-CN.md [hunk 3]
```diff
@@ -42,4 +42,5 @@ return (
 | csp | 设置 [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) 配置 | { nonce: string } | - | 3.13.1 |
 | renderEmpty | 自定义组件空状态。参考 [空状态](/components/empty/) | Function(componentName: string): ReactNode | - | 3.12.2 |
 | getPopupContainer | 弹出框（Select, Tooltip, Menu 等等）渲染父节点，默认渲染到 body 上。 | Function(triggerNode) | () => document.body | 3.11.0 |
+| locale | 语言包配置，语言包可到 [antd/lib/locale](http://unpkg.com/antd/lib/locale/) 目录下寻找 | object | - | 3.21.0 |
```
Comment (afc163) [inline components/config-provider/index.zh-CN.md]: `antd/es/locale` 吧。

File: components/locale-provider/index.tsx [hunk 4]
```diff
@@ -3,6 +3,9 @@ import * as PropTypes from 'prop-types';
 import * as moment from 'moment';
 import interopDefault from '../_util/interopDefault';
 import { ModalLocale, changeConfirmLocale } from '../modal/locale';
+import warning from '../_util/warning';
+
+export const ANT_MARK = 'internalMark';
```
Comment (afc163) [inline components/locale-provider/index.tsx]: 是不是可以用 Symbol

File: components/locale-provider/index.tsx [hunk 5]
```diff
@@ -48,6 +52,12 @@ export default class LocaleProvider extends React.Component<LocaleProviderProps,
     super(props);
     setMomentLocale(props.locale);
     changeConfirmLocale(props.locale && props.locale.Modal);
+
+    warning(
+      props._ANT_MARK__ === ANT_MARK,
+      'LocaleProvider',
+      '`LocaleProvider` is deprecated. Please use `locale` with `ConfigProvider` instead.',
```
Comment (afc163) [inline components/locale-provider/index.tsx]: 用 u.ant.design 给个链接。

File: components/locale-provider/index.tsx (hunk 4 above)
Comment (zombieJ) [reply components/locale-provider/index.tsx]: IE 不支持 Symbol

File: components/locale-provider/index.tsx (hunk 5 above)
Comment (zombieJ) [reply components/locale-provider/index.tsx]: 直接给个 ConfigProvider Demo 地址好了，复用一下。

File: components/locale-provider/index.tsx [hunk 6]
```diff
@@ -56,7 +56,7 @@ export default class LocaleProvider extends React.Component<LocaleProviderProps,
     warning(
       props._ANT_MARK__ === ANT_MARK,
       'LocaleProvider',
-      '`LocaleProvider` is deprecated. Please use `locale` with `ConfigProvider` instead.',
+      '`LocaleProvider` is deprecated. Please use `locale` with `ConfigProvider` instead: http://ant.design/components/config-provider-cn/#components-config-provider-demo-locale',
```
Comment (afc163) [inline components/locale-provider/index.tsx]: http://github.com/ant-design/u.ant.design

Comment (zombieJ) [reply components/locale-provider/index.tsx]: 懂了，thx~

**Comment by zombieJ** [comment]
Done. Please check. @afc163

File: components/locale-provider/index.tsx (hunk 4 above)
Comment (afc163) [reply components/locale-provider/index.tsx]: 不是有 Babel 转过

**Review by afc163 — APPROVED** [review]

**Pull request merged**